### PR TITLE
Decorate drive inventory objects, not connectors

### PR DESCRIPTION
**Pull request by amboar**
Fixes SW550924

### CODE REVIEW

File: src/inventory/inventory-manager.cpp [hunk 1]
```diff
@@ -26,13 +30,80 @@ static constexpr auto INVENTORY_MANAGER_OBJECT =
...
+                        break;
+                }
+            }
+        }
+    }
+    catch (const sdbusplus::exception::exception& ex)
+    {
+        warning(
+            "Failed to fetch inventory objects for migration: {EXCEPTION_DESCRIPTION}",
+            "EXCEPTION_DESCRIPTION", ex.what());
```
Comment (spinler) [inline src/inventory/inventory-manager.cpp:92]: with lg2 you can just pass in 'ex' directly without the .what if you wanted.

File: src/inventory/migrations/nvme-i2c-endpoint-from-slot-to-drive.cpp [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+#include <phosphor-logging/lg2.hpp>
+
+#include <filesystem>
+
+PHOSPHOR_LOG2_USING;
+
+using namespace inventory;
+
+Migration::Result MigrateNVMeI2CEndpointFromSlotToDrive::migrate(
```
Comment (spinler) [inline src/inventory/migrations/nvme-i2c-endpoint-from-slot-to-drive.cpp:14]: nit: To me a migration implies moving something, but this just does a delete?

File: src/tests/test-inventory-migrations.cpp [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+    MockInventory::accumulate(inventory.store, path, slot);
+
+    Inventory::migrate(&inventory, MigrateNVMeIPZVPDFromSlotToDrive());
+
+    std::map<std::string, inventory::ObjectType> untouchedSlot = {{path, slot}};
+
+    EXPECT_EQ(untouchedSlot, inventory.store);
+}
+
+TEST(MigrateNVMeIPZVPDFromSlotToDrive, matchedVPDDecoraion)
```
[the remark this answers is not in the document]
Comment (amboar) [reply src/tests/test-inventory-migrations.cpp:90]: Oh, thanks

**Comment by amboar** [comment]
> > if you had wanted, inventory-manager has a YAML file (inventory-cleanup.yaml) where you can put in path+interfaces of objects you'd like it to delete.
> 
> Oh, I did look, but could only see the ability to delete entire objects, not specific interfaces. It did surprise me. I'll follow up on that

Reflecting on this further, using the cleanup yaml file would require more coordination do it feels like less trouble this way even if it can't actually delete the interfaces

File: src/inventory/inventory-manager.cpp (hunk 1 above)
Comment (amboar) [reply src/inventory/inventory-manager.cpp:92]: Ah, neat. I'll fix that.

**Comment by amboar** [comment]
@spinler on the basis that you've approved this and my reasoning above, I'm going to fix up the nitpicks you pointed out above, re-push here then merge the PR. As a second step I'll send a patch for PIM to cleanup the interfaces on the slot objects.

**Pull request merged**